### ISSUE TITLE
Debug Logs Excessively Large

### ISSUE THREAD

**Issue by kjcjohnson** [veracode/gen-ir#56]
I attempted to generate a debug log today for a large iOS app, and it's on the order of 25GB...considering the [Veracode File Exchange](https://docs.veracode.com/r/using_exchangefile) only supports files up to 2GB, this seems not helpful. I'm sure there's lots of good stuff in it, but this seems too large to handle.

Is it possible to get more focused debug logging, say for a specific target? That should reduce the size and still (hopefully) provide useful information.

Command I'm running (on a CI server):
```sh
set -o pipefail && /opt/homebrew/opt/gen-ir@0.5.0-alpha/bin/gen-ir /Users/<redacted>/Logs/App-Build.log ./build/Products/App.xcarchive --project-path /Users/<redacted>/App.xcworkspace --debug | tee /Users/<redacted>/Logs/gen-ir.log | grep 'Operating on target:' --line-buffered
```

Note that the `App-Build.log` from xcodebuild is on the order of ~150MB, and `gen-ir` reports processing on the order of 11200 modules. Is the size of this debug log expected to be so large?

**Comment by buzzcode** [comment]
Hello.

Is there a problem you're trying to solve?  Often just looking at the end of the debug log will have a useful error message.

And while we don't support target-specific debugging today, that's an interesting idea.  Although how that might interact with target renaming (e.g., via a .xcconfig file) might add a wrinkle.

Dropping the compiler commands might also be a good enhancement to debugging (i.e., don't have the compiler commands displayed at "debug-level-1")

**Comment by kjcjohnson** [comment]
Yes, I am currently investigating why some bitcode files are ending up in the wrong IR folders in the .xcarchive, so I am attempting doing my own filtering of the log. In this case, `gen-ir` is actually succeeding, so there isn't anything useful I can find at just the end of the log file.

But also, one of the things that Veracode Support asks for when we submit a ticket is the log file from `gen-ir --debug`, so it would be nice to have an answer for them - whether or not that answer is "no, too big", or some other more focused information.

**Comment by davdres** [comment]
Since this issue was opened some of the logging has been trimmed.  Also, support should be able to work with you to provide files that are larger than 2GB.  If you are still having problems with bitcode files in the wrong folders, please open a new issue or contact support.
